addons/wowl/static/src/views/view.ts

```diff
@@ -0,0 +1,115 @@
+import { Component, tags, hooks } from "@odoo/owl";
+import { useService } from "../core/hooks";
+import { ActionRequest } from "../services/action_manager/action_manager";
+import { ViewDescription } from "../services/view_manager";
+import { FieldDefinition, OdooEnv, ViewProps, ViewType } from "../types";
+import { ControlPanelSubTemplates } from "./abstract_controller";
+import { Model, ModelParams } from "./model";
+import { FavoriteMenu } from "./favorite_menu/favorite_menu";
+import { Dialog } from "../components/dialog/dialog";
+import { Dropdown } from "../components/dropdown/dropdown";
+import { DropdownItem } from "../components/dropdown/dropdown_item";
+const { useSubEnv } = hooks;
+
+export interface Fields {
+  [fieldName: string]: FieldDefinition;
```

Comment:
FieldDefinition could be a union of "field type" in order to have better control and do not have optional parameters (similar to too loose rng validation)